search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/GroupFilterTransformer.java

```diff
@@ -21,7 +21,8 @@ public SearchQuery toSearchQuery(final GroupFilter filter) {
 
     return and(
         filter.groupKey() == null ? null : term("key", filter.groupKey()),
-        filter.name() == null ? null : term("name", filter.name()));
+        filter.name() == null ? null : term("name", filter.name()),
+        filter.memberKey() == null ? null : term("memberKey", filter.memberKey()));
```

Comment:
We need to use a `has_child` query now. We also need to add a filter to exclude the child documents from the result. There's an example in my PR for role members: https://github.com/camunda/camunda/pull/24609/files#diff-06612ab14d30600a4a3a4745a5d6605ad0a28c4217223ee9d4bd529a856b5e8c